internal/graph/object_providers.go

```diff
@@ -124,65 +111,72 @@ func (c *recursiveUsersetObjectProvider) End() {
 	if c.cancel != nil {
 		c.cancel()
 	}
-	if c.pool != nil {
```

Comment:
Again, there is a change in behaviour in that we don't wait for goroutine to be done. Please verify this is intentional

Reply:
Intentional.